src/server.c

```diff
@@ -6117,6 +6117,8 @@ void createPidFile(void) {
     if (fp) {
         fprintf(fp,"%d\n",(int)getpid());
         fclose(fp);
+    } else {
+        serverLog(LL_WARNING, "Failed to write PID file: %s", strerror(errno));
```

Comment:
Is it necessary to fail to start Redis when the creation of the pid file fails?

Reply:
That would be a backwards breaking change if someone upgrade and suddenly fails to start.